main/res/values/themes.xml

```diff
@@ -54,54 +50,51 @@
         <item name="android:textColor">@color/just_white</item>
     </style>
 
-    <style name="cgeo.ActionBarStyle.Light.Inverse" parent="@style/Widget.AppCompat.Light.ActionBar.Solid.Inverse">
-        <item name="indeterminateProgressStyle">@style/cgeo.Widget.AppCompat.Base.ProgressBar.Medium</item>
-        <item name="android:indeterminateProgressStyle" tools:ignore="NewApi">@style/cgeo.Widget.AppCompat.Base.ProgressBar.Medium</item>
-    </style>
-
-    <style name="cgeo.base" parent="@style/Theme.AppCompat">
-
-       <!-- For some reason we get a different text-size here (bug in abc?), explicitly set text styles -->
-       <item name="titleTextStyle">@style/TextAppearance.AppCompat.Widget.ActionMode.Title</item>
-       <item name="subtitleTextStyle">@style/TextAppearance.AppCompat.Widget.ActionMode.Subtitle</item>
-
-       <item name="android:buttonStyle">@style/button</item>
-       <item name="android:editTextStyle">@style/edittext</item>
-       <item name="android:windowContentOverlay">@null</item>
-       <item name="actionBarStyle">@style/cgeo.ActionBarStyle</item>
-
-        <item name="colorAccent">@color/colorAccent</item>
-
+    <style name="cgeo.base" parent="@style/Theme.AppCompat.DayNight.DarkActionBar">
```

Comment:
Does it make an optical difference if you use the even more modern `Theme.MaterialComponents.DayNight` (AFAIK based on `AppCompat`) instead? Cause https://material.io/ provides some really nice additional features and would help c:geo to become more attractive in the future...

Reply:
Haven't tried yet. Will it work even for old Android versions (API 20+)? I thought that is what `AppCompat` is meant for.

Reply:
Yes, `MaterialComponents` is in fact just a better styled variant of `AppCompat`. But it might also look weird in the c:geo UI. But it's worth to try...

Reply:
For some of the Material.io components, their themes are necessary since they add some new, requires attributes. 

For the case of apps wanting to stay with the layout base of appcompat, Material.io offers bridge themes though. Can we use one of those?

Those would be of form:
`Theme.MaterialComponents.DayNight.[...].Bridge`, e.g. in this example `Theme.MaterialComponents.DayNight.DarkActionBar.Bridge`

Reply:
Haven't seen those themes "in action" yet, but I guess we can try to use an up-to-date look. I'm worrying about technical backward-compatibility - which of the Material themes will run correctly under Android 5? Need to read more about it yet...

My next step would then be to merge the approaches from @eddiemuc used for the new filter and what I have done so far in this PR, and try to get things into master for further collaborate development. Then we still have settings using the old approach, but we can migrate them later, as soon as we have migrated them to `PreferenceFragment`.

Reply:
from https://material.io/develop/android/docs/getting-started:
> Doing an app-wide migration by changing your app theme to inherit from a Material Components theme is the recommended approach. However, be sure to test thoroughly afterwards, as components in existing layouts may change their looks and behavior.
---
>### Ensure you are using AppCompatActivity
>Using AppCompatActivity will ensure that all the components work correctly. If you are unable to extend from AppCompatActivity, update your activities to use AppCompatDelegate. This will enable the AppCompat versions of components to be inflated among other important things.
---

> I'm worrying about technical backward-compatibility - which of the Material themes will run correctly under Android 5?

The Material themes are using the whole AppCompat stuff. There are towsends of Android Apps already using googles material design themes out there, and many of them are even still supporting Android 4 or even lower... So backward-compatibility is definitely no problem...

---
>The following is the list of Material Components themes you can use to get the latest component styles and theme-level attributes.
>
> -    Theme.MaterialComponents
>-    Theme.MaterialComponents.NoActionBar
>-    Theme.MaterialComponents.Light
>-    Theme.MaterialComponents.Light.NoActionBar
>-    Theme.MaterialComponents.Light.DarkActionBar
>
>  [And one of these `DayNight` themes we should choose:]
>-    Theme.MaterialComponents.DayNight
>-    Theme.MaterialComponents.DayNight.NoActionBar
>-    Theme.MaterialComponents.DayNight.DarkActionBar

Reply:
Thanks @fm-sys and @eddiemuc for the additional insights. I'll continue with updating the base class to Material based theme.